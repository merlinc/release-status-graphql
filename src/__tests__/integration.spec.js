const { createTestClient } = require('apollo-server-testing');
// eslint-disable-next-line import/no-extraneous-dependencies
const gql = require('graphql-tag');
const createServer = require('./utils');

// const nock = require('nock');

// const {constructTestServer} = require('./__utils');

// the mocked REST API data
// const {mockLaunchResponse} = require('../datasources/__tests__/launch');
// the mocked SQL DataSource store
// const {mockStore} = require('../datasources/__tests__/user');

const GET_STATUS = gql`
  query status($org: String!, $project: String!) {
    status(org: $org, project: $project) {
      project
      commits {
        message
        sha
        url
        commits {
          sha
          author {
            date
            name
            email
          }
          committer {
            date
            name
            email
          }
          message
        }
        promotions {
          buildId
          env
          rough
          status
          timestamp
          url
        }
      }
    }
  }
`;

describe('Queries', () => {
  it('fetches list of statuses', async () => {
    // create an instance of ApolloServer that mocks out context, while reusing
    // existing dataSources, resolvers, and typeDefs.
    // This function returns the server instance as well as our dataSource
    // instances, so we can overwrite the underlying fetchers
    const server = createServer({
      context: () => ({ org: 'merlinc', project: 'release-status-testbed' })
    });

    // use our test server as input to the createTestClient fn
    // This will give us an interface, similar to apolloClient.query
    // to run queries against our instance of ApolloServer
    const { query } = createTestClient(server);
    const res = await query({
      query: GET_STATUS,
      variables: { org: 'merlinc', project: 'release-status-testbed' }
    });
    expect(res).toMatchSnapshot({
      extensions: expect.any(Object)
    });
  });
});
