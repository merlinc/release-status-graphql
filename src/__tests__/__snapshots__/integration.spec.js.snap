// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries fetches list of statuses 1`] = `
Object {
  "data": Object {
    "status": Object {
      "commits": Array [
        Object {
          "commits": null,
          "message": "Feat/add circleci (#3)

* Initial circleci config",
          "promotions": Array [
            Object {
              "buildId": 3,
              "env": "build",
              "rough": false,
              "status": null,
              "timestamp": "2020-01-01T23:21:21.104Z",
              "url": "https://circleci.com/gh/merlinc/release-status-testbed/3",
            },
          ],
          "sha": "3c5d0a3922accba42182c3fc647ebca52419af6e",
          "url": "https://api.github.com/repos/merlinc/release-status-testbed/commits/3c5d0a3922accba42182c3fc647ebca52419af6e",
        },
        Object {
          "commits": null,
          "message": "Merge pull request #1 from merlinc/add-node

Add package.json",
          "promotions": Array [],
          "sha": "079df5bd5f46bf4a243879cca55faf2e88e2b67d",
          "url": "https://api.github.com/repos/merlinc/release-status-testbed/commits/079df5bd5f46bf4a243879cca55faf2e88e2b67d",
        },
        Object {
          "commits": null,
          "message": "Update package.json",
          "promotions": Array [],
          "sha": "4c7460f3a286b5eebdc6f93394c1dcbc103b6e9d",
          "url": "https://api.github.com/repos/merlinc/release-status-testbed/commits/4c7460f3a286b5eebdc6f93394c1dcbc103b6e9d",
        },
        Object {
          "commits": null,
          "message": "Add package.json",
          "promotions": Array [],
          "sha": "9ac2d7c11db9ff386ac0408df149c61523caf081",
          "url": "https://api.github.com/repos/merlinc/release-status-testbed/commits/9ac2d7c11db9ff386ac0408df149c61523caf081",
        },
        Object {
          "commits": null,
          "message": "Initial commit",
          "promotions": Array [],
          "sha": "a21c62481031c3af80e67f10491fd4fd9fafd60a",
          "url": "https://api.github.com/repos/merlinc/release-status-testbed/commits/a21c62481031c3af80e67f10491fd4fd9fafd60a",
        },
      ],
      "project": "release-status-testbed",
    },
  },
  "errors": undefined,
  "extensions": Any<Object>,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
